tests/cmdline/commands/test_computer.py

```diff
@@ -146,13 +146,13 @@ def test_mixed(self):
         result = self.cli_runner.invoke(computer_setup, options, input=user_input)
         self.assertIsNone(result.exception, msg='There was an unexpected exception. Output: {}'.format(result.output))
 
-        new_computer = orm.Computer.objects.get(name=label)
+        new_computer = orm.Computer.objects.get(label=label)
         self.assertIsInstance(new_computer, orm.Computer)
 
         self.assertEqual(new_computer.description, options_dict_full['description'])
         self.assertEqual(new_computer.hostname, options_dict_full['hostname'])
-        self.assertEqual(new_computer.get_transport_type(), options_dict_full['transport'])
-        self.assertEqual(new_computer.get_scheduler_type(), options_dict_full['scheduler'])
+        self.assertEqual(new_computer.transport_type, options_dict_full['transport'])
+        self.assertEqual(new_computer.scheduler_type, options_dict_full['scheduler'])
         self.assertEqual(new_computer.get_mpirun_command(), options_dict_full['mpirun-command'].split())
         self.assertEqual(new_computer.get_shebang(), options_dict_full['shebang'])
         self.assertEqual(new_computer.get_workdir(), options_dict_full['work-dir'])
```

Comment:
What about these - shouldn't these also become properties?

Reply:
When making these changes I have followed the following:
> Rule of thumb here is that if the attribute corresponds to a column on the underlying
database model, a property is used.

Since these properties here are stored in the `metadata` column, and in a sense are comparable to a node's attributes, I am not sure if we should give properties for all of them. This is an arbitrary choice ultimately, but one that at least tries to keep consistency with the other entities such as the `Node` class. The database columns are exposed through properties, while anything else gets normal methods. In the latter, we should even ask ourselves to what level we want to give individual methods to all these sub attributes. Giving too many clutters the interface, however, it does make it tab-discoverable. Putting them behind a generic `Computer.get_property('shebang')` makes the interface cleaner and more maintainable but maybe less discoverable. I am not sure yet what the best solution is here